src/subs/FitSubroutines.cpp

```diff
@@ -1650,6 +1651,23 @@ void readColors(const img::FTable &extensionTable,
     }  // end loop over catalogs to read
 }
 
+// Get color information from an input vector of colors matched to the current matches.
+template <class S>
+void readColors(typename S::MCat &matches,
+                const std::vector<int> matchIDs,
```

Comment:
Why aren't these vectors passed by reference?

Reply:
Poor coding practice I guess? I changed this.